### PR TITLE
fix(api-extractor): Don't trim imports from external packages based on their release tags

### PR DESCRIPTION
**Pull request by Josmithr**
The original algorithm trimmed imports from external libraries when generating rollups based on the same release tag semantics of the importing library. But if a library chooses to use an external library's types in its own API, it doesn't matter how that external library has tagged its members.

Note that this **does not** have anything to do with `bundledPackages` behavior.

Fix for #4425.

## Example (single package)

Consider a library with 2 exports.
```typescript
/**
 * @beta
 */
export interface Foo {}

/**
 * @public
 */
export interface Bar extends Foo {}
```

The resulting `@public` rollup for the library would look something like:

```typescript
interface Foo {}

export interface Bar extends Foo {}
```

## Example (2 packages)

Now consider the scenario where `Foo` and `Bar` live in separate packages.

Package `foo` exports interface "Foo" (tagged as `@alpha`), and package `bar` exports `Bar` (tagged as `@public`).

The existing behavior would yield a `@[public` rollup for package `bar` like the following:

```typescript
export interface Bar extends Foo {}

/* Excluded from this release type: Foo */
```

Note that there is no import statement for Foo, resulting in an invalid rollup.

Regardless of whether or not `bar` exports the `Foo` interface, or how `Foo` is tagged, a declaration for it must exist in the rollup. In the single-package example, this is accomplished via an inline **non-exported** declaration of `Foo`. In the 2 package case, this must be done using an import statement, which must not be trimmed out.

### CODE REVIEW

**Comment by octogonz** [comment]
Based on our conversation today, the original algorithm seems wrong, and your fix is probably correct and unlikely to cause regressions.

File: apps/api-extractor/src/generators/DtsRollupGenerator.ts [hunk 1]
```diff
@@ -105,18 +105,9 @@ export class DtsRollupGenerator {
     // Emit the imports
     for (const entity of collector.entities) {
       if (entity.astEntity instanceof AstImport) {
+        // BUG MITIGATION: don't trim imports based on their release tags
```
Comment (Josmithr) [inline apps/api-extractor/src/generators/DtsRollupGenerator.ts:108]:
```suggestion
        // Note: it isn't valid to trim imports based on their release tags.
        // E.g. class Foo (`@public`) extends interface Bar (`@beta`) from some external library.
        // API-Extractor cannot trim `import { Bar } from "externa-library"` when generating its public rollup,
        // or the export of `Foo` would include a broken reference to `Bar`.
```

File: common/changes/@microsoft/api-extractor/dont-trim-imports_2024-02-21-00-32.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-extractor",
+      "comment": "fix: Don't trim imports from external packages based when generating rollups",
```
Comment (octogonz) [inline common/changes/@microsoft/api-extractor/dont-trim-imports_2024-02-21-00-32.json:5]:
```suggestion
      "comment": "Fix an issue where imports were trimmed from external packages based when generating .d.ts rollups",
```
https://rushjs.io/pages/best_practices/change_logs/

**Review by octogonz — APPROVED** [review]

**Comment by octogonz** [comment]
🚀  Released with API Extractor 7.42.0

**Pull request merged**